test/node-api/test_threadsafe_function/binding.c

```diff
@@ -168,20 +181,21 @@ static void join_the_threads(napi_env env, void *data, void *hint) {
       napi_call_function(env, undefined, js_cb, 0, NULL, NULL));
   NAPI_CALL_RETURN_VOID(env, napi_delete_reference(env,
       the_hint->js_finalize_cb));
+  if (alt_ref) {
+    NAPI_CALL_RETURN_VOID(env, napi_delete_reference(env, alt_ref));
+    alt_ref = NULL;
+  }
 }
 
 static napi_value StartThreadInternal(napi_env env,
-                                      napi_callback_info info,
+                                      napi_value* argv,
                                       napi_threadsafe_function_call_js cb,
                                       bool block_on_full) {
```

Comment:
Would it be possible to reduce code duplication by returning the original signature and adding a new `bool alt_ref_js_cb` parameter which would be used like
```C
if (alt_ref_js_cb) {
  NAPI_CALL(env, napi_create_reference(env, argv[0], 1, &alt_ref));
  argv[0] = NULL;
}
```
and then proceeding normally with the creation of the thread-safe function?